railties/test/test_unit/reporter_test.rb

```diff
@@ -210,4 +210,8 @@ def skipped_test
       st.time = 10
       st
     end
+
+    def test_run_command_regex
+      %r{bin/rails test|bin/test}
```

Comment:
Looking at the code again, I'm thinking of two things:
- I could use `setup` and `teardown` to set `Rails::TestUnitReporter.executable` to the value of my choice then restore it. This would make the output consistent across different ways to run the tests. However, when a test legitimately fails, it would display an incorrect way to be rerun. [This is already the case on CI](https://buildkite.com/rails/rails/builds/112300#01925d06-ea29-4bb5-8b42-01f56d6409dd/1301-1314) because it uses the default value set [here](https://github.com/rails/rails/blob/ff42150d2f2c8df2b9fe21d97fea23fef3810949/railties/lib/rails/test_unit/reporter.rb#L9):
    ```
    # Running:

    .F

    Failure:
    TestUnitReporterTest#test_outputs_colored_failed_results [test/test_unit/reporter_test.rb:156]:
    Expected /\e\[31mF\e\[0m\n\n\e\[31mFailure:\nTestUnitReporterTest::ExampleTest#woot \[test_unit\/reporter_test.rb:\d+\]:\nboo\n\e\[0m\n\n(?-mix:bin\/rails test|bin\/test) .*test\/test_unit\/reporter_test.rb:\d+\n\n/ to match "\e[31mF\e[0m\n\n\e[31mFailure:\nTestUnitReporterTest::ExampleTest#woot [test/test_unit/reporter_test.rb:153]:\nboo\n\e[0m\n\nbin/rails test test/test_unit/reporter_test.rb:9\n\n".

    bin/rails test test/test_unit/reporter_test.rb:150
    ⬆️ railties/bin/rails does not exist
    ```
    https://github.com/rails/rails/blob/ff42150d2f2c8df2b9fe21d97fea23fef3810949/railties/lib/rails/test_unit/reporter.rb#L9

- I could retrieve the set value for `Rails::TestUnitReporter.executable` and expect the output to contain that string:
    ```rb
        def test_run_command_regex # well the method name would need to change too since this is not a regex anymore
          Rails::TestUnitReporter.executable
        end
    ```